esp-hal/src/i2c.rs

```diff
@@ -455,6 +458,13 @@ where
     ) -> Self {
         crate::into_ref!(i2c, sda, scl);
 
+        PeripheralClockControl::reset(match i2c.i2c_number() {
```

Comment:
we should consider resetting the peripheral in all drivers - this makes it possible to easily and reliably recover from errors by re-creating the driver